services/hooks/src/listeners.js

```diff
@@ -117,39 +117,78 @@ class HookListeners {
     }
   }
 
-  /** Add / Remove bindings from he queue */
+  /** Add / Remove bindings from the queue and return the set of bindings that
+   * are in place.  This may differ from newBindings if, for example, an exchange
+   * does not exist or a routingKeyPattern is invalid.
+   */
   async syncBindings(queueName, newBindings, oldBindings) {
     // for direct AMQP operations, we need the full name of the queue (with the
     // queue/<namespace> prefix)
     const fullQueueName = this.client.fullObjectName('queue', queueName);
-    const result = [...oldBindings];
+    let result = [...oldBindings];
 
     if (this.client.isFakeClient) {
       // update the bindings for the FakePulseConsumer
       this.listeners[queueName].setFakeBindings(newBindings);
-    } else {
-      let intersection = _.intersectionWith(oldBindings, newBindings, _.isEqual);
-      const delBindings = _.differenceWith(oldBindings, intersection, _.isEqual);
-      const addBindings = _.differenceWith(newBindings, intersection, _.isEqual);
-      if (!addBindings.length && !delBindings.length) {
-        return;
-      }
-      debug(`${queueName}: updating bindings to ${JSON.stringify(newBindings)}`);
-      await this.client.withChannel(async channel => {
-        // perform unbinds first; these do not fail if the binding doesn't exist,
-        // making them idempotent.
-        for (let {exchange, routingKeyPattern} of delBindings) {
+      return newBindings;
+    }
+
+    let intersection = _.intersectionWith(oldBindings, newBindings, _.isEqual);
+    const delBindings = _.differenceWith(oldBindings, intersection, _.isEqual);
+    const addBindings = _.differenceWith(newBindings, intersection, _.isEqual);
+    if (!addBindings.length && !delBindings.length) {
+      return newBindings;
+    }
+    debug(`${queueName}: updating bindings to ${JSON.stringify(newBindings)}`);
```

Comment:
Would it make sense for this eventually to be `monitor.debug` lines?

Reply:
Yes.  I didn't really want to dip my foot in the water of changing debug logging to monitor logging..